RadarSDK/RadarLocationPremissionsManager.m

```diff
@@ -0,0 +1,167 @@
+                self.status = [[RadarLocationPermissionsStatus alloc] initWithStatus: kCLAuthorizationStatusAuthorizedAlways
+                                                            backgroundPopupAvailable:NO
+                                                                   inForegroundPopup:NO
+                                                   userRejectedBackgroundPermissions:NO];
+            }
+        }
+       
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(applicationWillResignActive)
+                                                     name:UIApplicationWillResignActiveNotification
+                                                   object:nil];
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(applicationDidBecomeActive)
+                                                     name:UIApplicationDidBecomeActiveNotification
+                                                   object:nil];        
+        
+    }
+    return self;
+}
+
+- (void)updateStatus:(RadarLocationPermissionsStatus *)status {
+    self.status = status;
+    [RadarLocationPermissionsStatus store:status];
+    [self sendDelegateUpdate];
+}
+
+- (void)sendDelegateUpdate {
+    if (@available(iOS 14.0, *)) {
+        [[RadarDelegateHolder sharedInstance] didUpdateLocationPermissionsStatus:self.status];
+    }
+}
+
+
+- (void)requestLocationPermissions:(BOOL)background {
+    if (background && self.status.locationManagerStatus == kCLAuthorizationStatusAuthorizedWhenInUse) {
+
+        danglingBackgroundPermissionsRequest = YES;
+
+        [self.locationManager requestAlwaysAuthorization];
+        if (@available(iOS 14.0, *)) {
+            RadarLocationPermissionsStatus *status = [[RadarLocationPermissionsStatus alloc] initWithStatus:self.locationManager.authorizationStatus
+                                                                                   backgroundPopupAvailable:NO
+                                                                                          inForegroundPopup:self.status.inForegroundPopup
+                                                                          userRejectedBackgroundPermissions: self.status.userRejectedBackgroundPermissions];
+            [self updateStatus:status];
+        }
+
+        // We set a flag that request has been made and start a timer. If we resign active we unset the timer.
+        // When the timer fires and the flag has not been unset, we assume that app never resigned active.
+        // Usually this means that the user has previously rejected the background permissions.
+        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.5 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+            if (self->danglingBackgroundPermissionsRequest) {
```

Comment:
why aren't we doing `self.danglingBackgroundPermissionsRequest`? We don't use this pointer notation elsewhere

Reply:
I made the decision to not expose `danglingBackgroundPermissionsRequest` as a property on the interface as it is an implementation detail. Hence it only exists as a instance variable and we cannot access it with `.`, only `->`. We can technically drop the `self->` but xcode's linter will be angry at us as this is within a `block`.

Reply:
actually changed my mind, apparently its not a pattern we use in the rest of the code, we seem to prefer just using interface and being more "heavy handed" and making stuff properties. Ultimately it's more a stylistic choice, defaulting to existing patterns in this case.